mlir/lib/Conversion/TosaToRock/TosaToRock.cpp

```diff
@@ -1720,7 +1863,25 @@ struct AttentionRewritePattern : public OpRewritePattern<tosa::MatMulOp> {
       rewriter.create<memref::CopyOp>(loc, resMemref, outMemref);
       rewriter.create<rock::YieldOp>(loc);
     }
-    rewriter.replaceOp(op, attnOp.getResult());
+    tosa::AddOp addOp;
+    Value expandedOutLse;
+    if (lse) {
+      // Reverse the collapse operation
+      SmallVector<ReassociationIndices> reassocIndices = {{0, 1}, {2, 3}};
+      expandedOutLse = rewriter.create<tensor::ExpandShapeOp>(
+          op.getLoc(), lseOrig.getType(), attnOp->getResult(1), reassocIndices);
+
+      // collecting AddOp before the first replace
+      addOp = lseOrig.getDefiningOp<tosa::AddOp>();
+
+      // all users have to be moved after the expand shape
+      moveUsersAfterExpandShape(rewriter, op.getLoc(),
```

Comment:
instead of moving all users after `expandedOutLse` wouldn't it better to insert expandedOutLse in correct location at first ? 

e.g. Find out `op.getLoc()` here

```
  expandedOutLse = rewriter.create<tensor::ExpandShapeOp>(
          op.getLoc(), lseOrig.getType(), attnOp->getResult(1), reassocIndices);
```